src/snowflake/cli/_plugins/nativeapp/entities/application.py

```diff
@@ -643,20 +669,14 @@ def create_or_upgrade_app(
         policy: PolicyBase,
         interactive: bool,
     ):
-        model = self._entity_model
-        console = self._workspace_ctx.console
-        debug_mode = model.debug
-
-        stage_fqn = stage_fqn or package.stage_fqn
-        stage_schema = extract_schema(stage_fqn)
-
         sql_executor = get_sql_executor()
+
         with sql_executor.use_role(self.role):
             event_sharing = EventSharingHandler(
-                telemetry_definition=model.telemetry,
+                telemetry_definition=self._entity_model.telemetry,
                 deploy_root=package.deploy_root,
                 install_method=install_method,
-                console=console,
+                console=self._workspace_ctx.console,
             )
```

Comment:
So about this use_role block :D
instantiating an EventSharingHandler shouldn't need to be in a use_role block. If it needs to be done with a specific role, pass the role in. 

Looking at implementation of `get_existing_app_info`, it seems that it could be moved as a whole to the facade. Just make sure you pass an optional role in, similar to other methods in the facade. 

For `_upgrade_app` and `_create_app`, you're already doing role handling in them.

It seems that this `use_role` is trivial now upon changes above and can be removed.